src/commands/deployWorkspaceProject/internal/DefaultResourcesNameStep.ts

```diff
@@ -4,83 +4,72 @@
-            cleanWorkspaceName(nonNullValueAndProp(context.rootFolder, 'name')))
-        );
-
-        const resourceBaseName: string = (await context.ui.showInputBox({
+export class DefaultResourcesNameStep extends AzureWizardPromptStep<DeployWorkspaceProjectInternalContext> {
+    public async prompt(context: DeployWorkspaceProjectInternalContext): Promise<void> {
+        const resourceName: string = (await context.ui.showInputBox({
             prompt: localize('resourceBaseNamePrompt', 'Enter a name for the new container app resource(s).'),
-            validateInput: this.validateInput,
+            validateInput: (name: string) => this.validateInput(context, name),
             asyncValidationTask: (name: string) => this.validateNameAvailability(context, name)
         })).trim();
 
-        ext.outputChannel.appendLog(localize('usingResourceName', 'User provided the new resource name "{0}" as the default for resource creation.', resourceBaseName))
-
-        !context.resourceGroup && (context.newResourceGroupName = resourceBaseName);
-        !context.managedEnvironment && (context.newManagedEnvironmentName = resourceBaseName);
-        !context.containerApp && (context.newContainerAppName = resourceBaseName);
-        context.imageName = ImageNameStep.getTimestampedImageName(context.containerApp?.name || resourceBaseName);
-    }
-
-    public async configureBeforePrompt(context: DeployWorkspaceProjectContext): Promise<void> {
-        const workspaceName: string = cleanWorkspaceName(nonNullValueAndProp(context.rootFolder, 'name'));
-        if (this.validateInput(workspaceName) !== undefined) {
-            context.telemetry.properties.promptDefaultNameReason = 'invalid';
-            return;
-        }
-
-        if (!await this.isWorkspaceNameAvailable(context, workspaceName)) {
-            context.telemetry.properties.promptDefaultNameReason = 'unavailable';
-            return;
-        }
-
-        if (!context.resourceGroup || !context.managedEnvironment || !context.registry || !context.containerApp) {
-            ext.outputChannel.appendLog(localize('usingWorkspaceName', 'Using workspace name "{0}" as the default for remaining resource creation.', workspaceName));
-        }
+        ext.outputChannel.appendLog(localize('usingResourceName', 'User provided the new resource name "{0}" as the default for resource creation.', resourceName))
 
-        !context.resourceGroup && (context.newResourceGroupName = workspaceName);
-        !context.managedEnvironment && (context.newManagedEnvironmentName = workspaceName);
-        !context.registry && (context.newRegistryName = await RegistryNameStep.tryGenerateRelatedName(context, workspaceName));
-        !context.containerApp && (context.newContainerAppName = workspaceName);
-        context.imageName = ImageNameStep.getTimestampedImageName(context.containerApp?.name || workspaceName);
+        !context.resourceGroup && (context.newResourceGroupName = resourceName);
```

Comment:
I still suggest we should add a suffix for each resource type.

`resourceName-rg`
`resourceName-me` or `resourceName-env`
`resourceName-app` or `resourceName-aca`

I think when you see them all in the resource group, it makes it a lot easier to distinguish what each resource is. I notice other tooling will do that as well.

Reply:
I like that idea, let me do that in a separate PR after we discuss what that convention should look like